packages/desktop-client/src/browser-preload.browser.js

```diff
@@ -89,7 +87,13 @@ global.Actual = {
             alert('Error reading file');
           };
         }
-      });
+      };
+
+      // In Safari the file input has to be in the DOM for change events to
+      // reliably fire.
+      document.body.appendChild(input);
```

Comment:
Since it really only makes sense to have one file dialog open at a time, could you add one input to the body the first time this method is called, then reuse it for future calls?

Reply:
Sounds good 👍 implemented in https://github.com/actualbudget/actual/pull/1349/commits/8b01e17e6251cbf7ec684f6c1cc0daea770630f5.